TileMap mouse right can delete autotile

magian1127 opened the pull request:
3.x has the function, 4.0 is not.
There are two changes. One is when setting, one is when using.

1
![image](https://user-images.githubusercontent.com/14800320/193257076-eb43d3e4-0f8b-42f9-a802-f6b3323448bf.png)

2
![image](https://user-images.githubusercontent.com/14800320/193256985-d21e2836-16ab-4330-b665-3febe71c2497.png)


TileMap鼠标右键可以删除自动瓷砖.
3.x有的功能,不知道为什么4.0没有了.
分别改了两处.一个是设置时,一个是使用时

KoBeWi commented:
I think the TileSet part is being implemented by #65903
But it's still needed for TileMap.

groud commented:
@akien-mga I already have a PR open for that: https://github.com/godotengine/godot/pull/65903

magian1127 commented:
I didn't find it after searching before... Next time I'll have to look more carefully...

KoBeWi commented:
Doesn't seem to work correctly. Deleting tiles causes more tiles to appear, so you can't remove them completely:

https://user-images.githubusercontent.com/2223172/193266487-ff6390c3-4368-441a-9d50-7972f386a9bb.mp4

Seems like right-clicking empty space creates new tiles.

Also this needs support for other tools (line, rectangle, fill).

magian1127 committed:
TileMap mouse right can delete autotile

magian1127 commented:
@KoBeWi 
Sorry, I found something wrong when I changed it just now, so I changed it into a draft. I let you test it in vain.
Now, I tested that several shapes can be deleted norma.

KoBeWi reviewed (approved):
Works fine.

Seems like the functionality was already there, but it was bugged. Needs a check from @groud

akien-mga reviewed (approved):
Makes sense to me.

akien-mga commented:
Thanks!
